python/mxnet/module/base_module.py

```diff
@@ -363,6 +363,19 @@ def predict(self, eval_data, num_batch=None, merge_batches=True, reset=True,
             eval_data.reset()
 
         output_list = []
+        # If data provided is a dict. This makes sure that only relevant data items
```

Comment:
Do we need to make this change for score too ?

Reply:
Yes, thanks for pointing this out. Added to score() too!!